vtree/diff.js

```diff
@@ -343,16 +348,23 @@ function reorder(aChildren, bChildren) {
 // TODO: We really want to make this more efficient
 // Finding the current index of any key in the array could be
 // faster.
-function moveItem(array, toIndex, itemKey) {
+function indexOfKey(array, itemKey) {
```

Comment:
the updating after a mutation is O(N) which is the same as not keeping the hash and scanning the list like we do now - the hash is technically worse (by just a little bit) since it requires more memory.